src/Sylius/Bundle/WebBundle/Controller/Frontend/Account/OrderController.php

```diff
@@ -140,15 +141,16 @@ protected function findOrderOr404($number)
             throw $this->createNotFoundException('The order does not exist.');
         }
 
-        if (!$this->get('security.context')->isGranted('ROLE_SYLIUS_ADMIN')
-            && (!$this->getCustomer() || $this->getCustomer()->getId() !== $order->getCustomer()->getId())
-        ) {
+        if (!$this->get('security.authorization_checker')->isGranted('ROLE_SYLIUS_ADMIN') && !$order->getCustomer()->isEqualTo($this->getCustomer())) {
             throw new AccessDeniedException();
         }
 
         return $order;
     }
 
+    /**
+     * @return null|CustomerInterface
```

Comment:
Rly? This means totally same, and even mine is more correct IMO as `null` is returned faster in context than object...